controllers/azuremanagedcontrolplane_controller_test.go

```diff
@@ -176,3 +182,82 @@ func TestAzureManagedControlPlaneReconcilePaused(t *testing.T) {
 	g.Expect(err).To(BeNil())
 	g.Expect(result.RequeueAfter).To(BeZero())
 }
+
+func TestAzureManagedControlPlaneReconcileNormal(t *testing.T) {
+	g := NewWithT(t)
+	ctx := context.Background()
+	cp := &infrav1.AzureManagedControlPlane{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "fake-azmp",
+			Namespace: "fake-ns",
+		},
+		Spec: infrav1.AzureManagedControlPlaneSpec{
+			AzureManagedControlPlaneClassSpec: infrav1.AzureManagedControlPlaneClassSpec{
+				Version: "0.0.1",
+			},
+		},
+		Status: infrav1.AzureManagedControlPlaneStatus{
+			Ready:       false,
+			Initialized: false,
+		},
+	}
+	scheme, err := newScheme()
+	g.Expect(err).ToNot(HaveOccurred())
+
+	client := fake.NewClientBuilder().WithScheme(scheme).WithObjects(cp).WithStatusSubresource(cp).Build()
+	amcpr := &AzureManagedControlPlaneReconciler{
+		Client: client,
+	}
+
+	helper, err := patch.NewHelper(cp, client)
+	g.Expect(err).ToNot(HaveOccurred())
+
+	scopes := &scope.ManagedControlPlaneScope{
+		Cluster: &clusterv1.Cluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "fake-cluster",
+				Namespace: "fake-ns",
+			},
+		},
+		Client:       client,
+		PatchHelper:  helper,
+		ControlPlane: cp,
+	}
+	scopes.SetAdminKubeconfigData(createFakeKubeConfig())
+	scopes.SetUserKubeconfigData(createFakeKubeConfig())
+
+	amcpr.getNewAzureManagedControlPlaneReconciler = func(scope *scope.ManagedControlPlaneScope) (*azureManagedControlPlaneService, error) {
+		ctrlr := gomock.NewController(t)
+		defer ctrlr.Finish()
+		svcr := mock_azure.NewMockServiceReconciler(ctrlr)
+		svcr.EXPECT().Reconcile(gomock.Any()).Return(nil).AnyTimes()
```

Comment:
I noticed that this test fails if I remove the `.AnyTimes()` here due to not calling `Reconcile()` the expected one time, which is wrong because I think the `ctrlr.Finish()` is executing too soon (at the end of this inner function vs. at the very end of the test).

I'd vote to remove the `AnyTimes()` or make an equivalent change that ensures we're calling that `Reconcile` at least once, and then fix the test failure by removing `defer ctrlr.Finish()` entirely, which isn't necessary when initializing the mock controller with a `*testing.T` like is already being done here.

Reply:
_You can tell I used to use the old gomock when writing tests, that I forgot_ (facepalm) 

Removing and pushing again.